spec/rubocop/cop/metrics/utils/abc_size_calculator_spec.rb

```diff
@@ -2,78 +2,184 @@
 
 RSpec.describe RuboCop::Cop::Metrics::Utils::AbcSizeCalculator do
   describe '#calculate' do
-    context '0 assignments, 3 branches, 0 conditions' do
-      it 'returns 3' do
-        node = parse_source(<<~RUBY).ast
-          def method_name
-            return x, y, z
+    subject(:vector) { described_class.calculate(node).last }
+
+    let(:node) { parse_source(source).ast }
+
+    context 'multiple calls with return' do
+      let(:source) { <<~RUBY }
+        def method_name
+          return x, y, z
+        end
+      RUBY
+
+      it { is_expected.to eq '<0, 3, 0>' }
+    end
+
+    context 'with ||=' do
+      let(:source) { <<~RUBY }
+        def method_name
+          x = nil
+          x ||= 1
+        end
+      RUBY
+
+      it { is_expected.to eq '<2, 0, 1>' }
+    end
+
+    context 'with &&=' do
+      let(:source) { <<~RUBY }
+        def method_name
+          x = nil
+          x &&= 1
+        end
+      RUBY
+
+      it { is_expected.to eq '<2, 0, 1>' }
+    end
+
+    context 'assignment with ternary operator' do
+      let(:source) { <<~RUBY }
+        def method_name
+          a = b ? c : d
+          e = f ? g : h
+        end
+      RUBY
+
+      it { is_expected.to eq '<2, 6, 2>' }
+    end
+
+    context 'with an empty block' do
+      let(:source) { <<~RUBY }
+        def method_name
+          x = foo    # <1, 1, 0>
+          bar {}     # <1, 2, 0>
+        end
+      RUBY
+
+      it { is_expected.to eq '<1, 2, 0>' }
+    end
+
+    context 'with a block' do
+      let(:source) { <<~RUBY }
+        def method_name
+          x = foo    # <1, 1, 0>
+          bar do     # <1, 3, 0>  (+1 for bar, +1 for non-empty block)
```

Comment:
I'm not so sure that a block in itself should count as a branch. It says in `abc_size_calculator.rb` that a branch is "an explicit forward program branch out of scope". So a code block that can be called from outside the scope of the current method is not a branch IMO.

Reply:
Yeah, I'm not so sure. I read "For example, the B (branch) count is virtually identical to "cyclomatic complexity" [here](https://wiki.c2.com/?AbcMetric), but I'm not sure I can really make sense of that. Maybe it would be best not to count it. OTOH, `yield` should be counted as branches and I don't think they are

Reply:
FWIW, maybe the quote meant to say "For example, the C (condition) count is virtually identical to "cyclomatic complexity". Maybe we should consider counting blocks as a "condition", in the sense that control flow is not linear at a block and entering it or not will be decided at runtime; exactly like and `if`s or `for`. What do you think?

Reply:
I agree on `yield`. About cyclomatic complexity and blocks, I believe that's one of those things we can't know for sure with static analysis. The block could be another possible path through the method, but then again it might not be.

Reply:
When I said "additional disconnected node" I was wrong. Could be disconnected, if the block is captured and executed later, but typically it will behave like a loop/condition and be connected to the graph and add +1. In all cases it adds +1 to the cyclomatic complexity though.

Reply:
For Cyclomatic complexity, https://github.com/metricfu/Saikuro also counts +1 for blocks. I just can't find an example of Abc calculator though